python_scripts/generate_test_data.py

```diff
@@ -0,0 +1,458 @@
+# This script generates test data for the database.
+# Check the /config/defaults contains your database connection information.
+# You may want to run the reset_database.py script first to create the tables,
+# views and stored procedures.
+
+# Set NUMBER_DESTINATION_PLATES_TO_MAKE below to the number of plates you want.
+# Run the script at the command line using:
+# python3 generate_test_data.py
+
+import datetime
+import os
+import mysql.connector # use command 'pip install mysql-connector-python'
+import json
+import random
+
+from config.defaults import *
+
+NUMBER_DESTINATION_PLATES_TO_MAKE = 1
+GBG_METHOD_NAME = 'example method name v1.2'
+USER_ID = 'ab12'
+SYSTEM_NAME = 'CPA'
+CONTROL_PLATE_BC = 'DN999999999'
+
+WELL_COORDS = [
+    "A1","A2","A3","A4","A5","A6","A7","A8","A9","A10","A11","A12",
+    "B1","B2","B3","B4","B5","B6","B7","B8","B9","B10","B11","B12",
+    "C1","C2","C3","C4","C5","C6","C7","C8","C9","C10","C11","C12",
+    "D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12",
+    "E1","E2","E3","E4","E5","E6","E7","E8","E9","E10","E11","E12",
+    "F1","F2","F3","F4","F5","F6","F7","F8","F9","F10","F11","F12",
+    "G1","G2","G3","G4","G5","G6","G7","G8","G9","G10","G11","G12",
+    "H1","H2","H3","H4","H5","H6","H7","H8","H9","H10","H11","H12"
+]
+
+# Create a connection to the database
+def create_database_connection():
+    db_conn = mysql.connector.connect(
+       host=DB_HOST,
+       port=DB_PORT,
+       user=DB_USER,
+       passwd=DB_PWD,
+       database=DB_NAME
+    )
+    return db_conn
+
+# Fetch the configuration details for a specific system
+def get_configuration_for_system(system_name) -> dict:
+    try:
+        # open DB connection
+        db_conn = create_database_connection()
+
+        # Get a cursor
+        cursor = db_conn.cursor(dictionary=True, buffered=True)
+
+        # Call stored procedure with system name as parameter
+        args = [system_name]
+        cursor.callproc('getConfigurationForSystem', args)
+
+        # Process the result into a dictionary
+        dict_result = {}
+        for result in cursor.stored_results(): # only one result
+            configs = result.fetchall() # array of key value pairs
+            for config in configs:
+                dict_result.update({config['config_key']: config['config_value']})
+
+        return dict_result
+
+    except Exception as e:
+        print(e)
+    else:
+        print("Fetched configuration for system name %s" % system_name)
+    finally:
+        cursor.close()
```

Comment:
check that cursor is not None (what happens if db_conn.cursor(...) fails, but we have a database connection

Reply:
Good point. Want to refactor out some of the repetition in this connection code too if there is time.